numpy/core/function_base.py

```diff
@@ -1,10 +1,24 @@
 from __future__ import division, absolute_import, print_function
 
+import warnings
+import operator
+
 __all__ = ['logspace', 'linspace']
 
 from . import numeric as _nx
 from .numeric import result_type, NaN, shares_memory, MAY_SHARE_BOUNDS, TooHardError
 
+def _index_deprecate(i, stacklevel=2):
```

Comment:
Don't use a function for this and using the `try ... except` construct seems unwieldy. I suspect you could either check for the `__index__` attribute, or use the `numbers` module
```
if not isinstance(i, numbers.Integral):
    deprecate...
i = int(i)
```